sqlx-mysql/src/types/str.rs

```diff
@@ -12,6 +13,7 @@ impl Type<MySql> for str {
         MySqlTypeInfo {
             r#type: ColumnType::VarString, // VARCHAR
             flags: ColumnFlags::empty(),
+            collation: Collation::utf8mb3_bin,
```

Comment:
@DrewMcArthur 

I think using `utf8mb3_bin` regardless of session state is misleading.
In my opinion, the only difference needed in this context is whether it is binary or string.
How about using an enum like this one?

```rust
CollationType {
    Binary,
    String
}

Reply:
This was one of the pieces I wasn't sure about, what the default should be for some of these.  I think you're right, `_bin` probably isn't right & I'll update this to `utf8mb3_general_ci` if you think that'd be correct.

As for changing the type of `MySqlTypeInfo`'s `collation` field to an enum, I think the source of the issue on #3387 is that there are three cases being compressed into the `ColumnFlag::binary` flag:

- a `VARCHAR` has a collation like `utf8_general_ci`, and the `ColumnFlag::binary` is false, so this is compatible with a Rust `String`
- a `BLOB` has a collation of `binary`, `ColumnFlag::binary` is true, so it's not compatible with a Rust `String`, which is also correct
- a `VARCHAR` has a collation of `*_bin`, so `ColumnFlag::binary` is true, so it's marked incompatible when we should be able to decode this into a `String`.  This is the case that needs to be fixed.

This could be handled with that enum, if the value is `String` for both `VARCHAR` cases, but I feel like that could also be confusing to have a `CollationType::String` (as opposed to binary) while simultaneously having `ColumnFlag::binary = true`. 

I think my preference is to keep track of each unique collation here, since it's only a byte, but I'll defer to you and @abonander to decide what makes more sense.

Reply:
```suggestion
            collation: Collation::utf8mb3_general_ci,
```

Reply:
^ @alu (sorry, I'm terrible at remembering to directly @ people in replies here)